### PR TITLE
Runtime Changes to Match the Latest QIR Spec

### PR DESCRIPTION
**Pull request by kuzminrobin**

### CODE REVIEW

File: src/QirRuntime/public/QirRuntime.hpp [hunk 1]
```diff
@@ -160,7 +162,6 @@ extern "C"
     // Initializes the callable with the provided function table and capture tuple. The capture tuple pointer
     // should be null if there is no capture.
     typedef void (*t_CallableEntry)(PTuple, PTuple, PTuple);                                                // NOLINT
-    typedef void (*t_CaptureCallback)(PTuple, int64_t);                                                     // NOLINT
     QIR_SHARED_API QirCallable* quantum__rt__callable_create(t_CallableEntry*, t_CaptureCallback*, PTuple); // NOLINT
```
Comment (kuzminrobin) [inline src/QirRuntime/public/QirRuntime.hpp:165]: This `typedef void (*t_CaptureCallback)(PTuple, int64_t)` was a duplicate. Removed.

File: src/QirRuntime/lib/QIR/bridge-rt.ll [hunk 2]
```diff
@@ -370,26 +386,46 @@ define dllexport %Callable* @__quantum__rt__callable_make_controlled(%Callable*
   ret %Callable* %.clb_cnt
 }
 
-define dllexport void @__quantum__rt__callable_memory_management(i32 %index, %Callable* %.clb, i64 %parameter) {
+; After both the compiler and runtime are in sync with https://github.com/microsoft/qsharp-language/pull/83:
```
Comment (swernli) [inline src/QirRuntime/lib/QIR/bridge-rt.ll:389]: I think this is ready to come out with this PR, right?

File: src/QirRuntime/test/FullstateSimulator/qsharp/qir-test-simulator.qs [hunk 3]
```diff
@@ -33,49 +33,49 @@ namespace Microsoft.Quantum.Testing.QIR
...
         if (res != 0) { return 60 + res; }
 
-        using((targets, ctls) = (Qubit[2], Qubit[2]))
+        use (targets, ctls) = (Qubit[2], Qubit[2])
         {
             let theta = 0.42;
             Exp([PauliX, PauliY], theta, targets);
             Adjoint Exp([PauliX, PauliY], theta, targets);
-            if (M(target) != Zero) { let res = 1; }
+            if (M(targets[0]) != Zero) { set res = 1; }
```
Comment (swernli) [inline src/QirRuntime/test/FullstateSimulator/qsharp/qir-test-simulator.qs:62]: As you pointed out in my change, this probably also include putting the rest of the scope body into an else clause.

**Review by swernli — APPROVED** [review]
A couple minor things, but otherwise looks good!

File: src/QirRuntime/lib/QIR/bridge-rt.ll (hunk 2 above)
Comment (kuzminrobin) [reply src/QirRuntime/lib/QIR/bridge-rt.ll:389]: Yes.

File: src/QirRuntime/test/FullstateSimulator/qsharp/qir-test-simulator.qs (hunk 3 above)
Comment (kuzminrobin) [reply src/QirRuntime/test/FullstateSimulator/qsharp/qir-test-simulator.qs:65]: Fixed. Thanks ;-)

**Comment by kuzminrobin** [comment]
https://teams.microsoft.com/l/message/19:bafaeb809d7342d68f4ad99cdc1675af@thread.skype/1616095642486?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=e6e22ac8-e11f-44d0-b599-46e9bea42c53&parentMessageId=1615235661210&teamName=Quantum%20Systems&channelName=QIR&createdTime=1616095642486

**Pull request merged**